includes/classes/SiteAutomation/EndPoints.php

```diff
@@ -0,0 +1,343 @@
+<?php
+/**
+ * EndPoints for curated content
+ *
+ * @package SophiWP
+ */
+
+namespace SophiWP\SiteAutomation;
+
+use WP_REST_Controller;
+use WP_REST_Server;
+use function SophiWP\Settings\get_sophi_settings;
+use function SophiWP\Blocks\SiteAutomationBlock\render_block_callback;
+
+/**
+ * Class: EndPoints
+ */
+class EndPoints extends WP_REST_Controller {
+
+	/**
+	 * Sophi default namespace for REST endpoints.
+	 */
+	const SOPHI_NAMESPACE = 'sophi/v1';
+
+	/**
+	 * Auth object.
+	 *
+	 * @var Request $auth
+	 */
+	public $auth;
+
+	/**
+	 * Request object.
+	 *
+	 * @var Request $request
+	 */
+	public $request;
+
+	/**
+	 * Class constructor.
+	 *
+	 */
+	public function __construct( $auth, $request ) {
+		$this->auth    = $auth;
+		$this->request = $request;
+
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Called automatically on `rest_api_init()`.
+	 */
+	public function register_routes() {
+		// GET /sophi/v1/site-automation route.
+		register_rest_route(
+			self::SOPHI_NAMESPACE,
+			'/site-automation',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'site_automation' ),
+					'permission_callback' => '__return_true',
+					'args'                => $this->site_automation_params(),
+				),
+			)
+		);
+
+		// POST /sophi/v1/update-posts route.
+		register_rest_route(
+			self::SOPHI_NAMESPACE,
+			'/site-automation-override',
+			array(
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'site_automation_override' ),
+					'permission_callback' => '__return_true',
```

Comment:
So it seems we check if a user is logged-in within our normal `callback`. Wondering if that would be bette within this `permission_callback`, rather than just returning `true` here?

Reply:
https://github.com/globeandmail/sophi-for-wordpress/pull/333/commits/9366ca91567a1e86603d0d324ec6727bce23ab21